src/pages/index.js

```diff
@@ -32,79 +39,167 @@ const profileDescriptionInput = document.querySelector(
-      const cardElement = renderCard(item);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-section.renderItems();
+/* -------------------------------------------------------------------------- */
+/*                               Form Validators                              */
+/* -------------------------------------------------------------------------- */
 
 const profileFormValidator = new FormValidator(settings, profileEditModal);
 profileFormValidator.enableValidation();
 
 const addFormValidator = new FormValidator(settings, addCardModal);
 addFormValidator.enableValidation();
 
-// Event listeners
-profileEditButton.addEventListener("click", () => {
-  const user = userInfo.getUserInfo();
-  profileTitleInput.value = user.name;
-  profileDescriptionInput.value = user.description;
-  modalEditForm.open();
-});
+const editProfilePicFormValidator = new FormValidator(
+  settings,
+  editProfileImageModal
+);
+editProfilePicFormValidator.enableValidation();
 
-addNewCardButton.addEventListener("click", () => {
-  addFormValidator._disableSubmitButton();
-  addCardImageModal.open();
+/* -------------------------------------------------------------------------- */
+/*                        Class Insantiation Constants                        */
+/* -------------------------------------------------------------------------- */
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "675ffc74-7010-43d4-a4e4-01622ad8dbf1",
+    "Content-Type": "application/json",
+  },
 });
 
-function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+/* -------------------------------------------------------------------------- */
+/*                            Profile Picture Edit                            */
+/* -------------------------------------------------------------------------- */
 
-  return card.getView();
+function loadInitialProfilePicture() {
+  api
+    .loadInitialProfileData()
+    .then((profileData) => {
+      profilePictureImage.src = profileData.avatar;
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+}
+
+loadInitialProfilePicture();
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Hanlders                               */
+/* -------------------------------------------------------------------------- */
+
+function handleEditProfileSubmit(formData) {
+  modalEditForm.setSubmitText(true);
+  api
+    .editProfileData(formData)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      modalEditForm.close();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => modalEditForm.setSubmitText(false));
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```